File: solutions/274/274.java
class Solution {
    public int hIndex(int[] citations) {
        // Sort the citations in descending order
        Arrays.sort(citations);

        int h_index = 0;

        for (int i = citations.length - 1; i >= 0; i--) {
            if (citations[i] >= citations.length - i) {
                h_index = citations.length - i;
            } else {
                break;
            }
        }

        return h_index;
    }
}

File: solutions/274/274.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        // Sort the citations in descending order
        sort(citations.begin(), citations.end(), greater<int>());
        
        int h_index = 0;

        for (int i = 0; i < citations.size(); i++) {
            if (citations[i] >= i + 1) {
                h_index = i + 1;
            }
        }

        return h_index;
    }
};